### DISCUSSION TITLE
Why is virtual_* used next to virtual_transport ?

### DISCUSSION THREAD

**Discussion by di-rect** [docker-mailserver/docker-mailserver#3463] [Q&A]
Hi,

Why is virtual_* used by Postfix while Dovecot is the LDA ?

Thanks

**Comment by polarathene** [comment]
Could you be less vague please?

https://github.com/docker-mailserver/docker-mailserver/blob/b001f5a14056fa4df5bf50466247f3f4e4d8000e/target/postfix/main.cf#L90-L94

That's the postfix default `main.cf` config we ship before our startup scripts run. `virtual_transport` is set to the Dovecot via LMTP. By default if the mail is not sent to a [domain value from `mydestination`](https://github.com/docker-mailserver/docker-mailserver/blob/b001f5a14056fa4df5bf50466247f3f4e4d8000e/target/postfix/main.cf#L15) (eg: hostname), it will go to Dovecot, otherwise Postfix handles it (_intended for system accounts like `clamav`, `amavis`, etc to send mail to_).

Usually this is for `postmaster@mail.example.com`, and the sender may be Postfix or one of the system accounts like `amavis@mail.example.com`. Sometimes mail may go to a system account instead (delivery failures IIRC, maybe other scenarios), In which case these are not handed over to Dovecot and get delivered to `/var/mail`. You don't want that, and we advise setting up `aliases` (_different from [virtual aliases](https://docker-mailserver.github.io/docker-mailserver/v12.1/config/user-management/#aliases) that we usually refer to as an alias in DMS_).

This Postfix system alias part is a bit of legacy detail that no ones had time to address any better and is mostly undocumented for DMS AFAIK. Common advice in past issues has been to alias `amavis` and any other users that drop mail into `/var/mail` directly from Postfix to redirect mail to the `root` user, and then have that `root` user alias to your [`POSTMASTER_ADDRESS`](https://docker-mailserver.github.io/docker-mailserver/v12.1/config/environment/#postmaster_address), which [we do document advice to add a virtual alias for](https://docker-mailserver.github.io/docker-mailserver/v12.1/usage/#aliases) (_with `setup alias add`_) to your own preferred account.

---

What is the actual concern?

The other `virtual_*` keys are related to information Postfix will use for knowing when to send mail that way.
